python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -984,8 +984,29 @@ def _get_relation_from_adapter(adapter: BaseAdapter, relation_key: RelationKey)
     )
 
 
+class DbtAttachMetadataContext(NamedTuple):
```

Comment:
better to use `@record`

Reply:
Should this be included as a top-level export?

Also we should probably add in the docstring that users shouldn't instantiate it themselves, to give ourselves freedom to change its constructor.

Also it should probably be experimental.

Also we should make sure any experimental warnings we raise for constructing it don't bubble up to users.

Also should some /all of its attributes and methods be public?

Reply:
Updated